modules/standard/IO.chpl

```diff
@@ -3267,6 +3267,29 @@ inline proc channel.read(ref args ...?k,
   return !error;
 }
 
+
+/*
+  Iterate over all of the lines in a channel.
+  Only serial iteration is supported.
+
+  :yields: lines read by the channel
+ */
+iter channel.lines() {
+  var local_style: iostyle;
+
+  // Update local_style to iterate over newlines
+  qio_channel_get_style(this._channel_internal, local_style);
+  local_style.string_format = QIO_STRING_FORMAT_TOEND;
+  local_style.string_end = 0x0a; // '\n'
+  qio_channel_set_style(this._channel_internal, local_style);
+
+  var lineReader = new ItemReader(string, this.kind, this.locking, this);
+  for line in this.itemReader(string, this.kind) {
+    yield line;
+  }
```

Comment:
To @vasslitvinov's comment, it should save style and then set it back again. so that the string_end change only applies to within this function. I'd also consider locking the channel for the duration of channel.lines(), to prevent any possible race conditions on the style.